resources/tests/test_resources_mutations.py

```diff
@@ -0,0 +1,284 @@
+    assert_field_missing,
+    assert_not_enough_permissions,
+    GraphQLTestClient,
+)
+from resources.models import Berth
+
+client = GraphQLTestClient()
+
+GRAPHQL_URL = "/graphql_v2/"
+
+
+CREATE_BERTH_MUTATION = """
+    mutation CreateBerth($input: CreateBerthMutationInput!) {
+      createBerth(input: $input) {
+        berth {
+          id
+          number
+          comment
+          pier {
+            id
+          }
+          berthType {
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+def test_create_berth(pier, berth_type, superuser):
+    pier_id = to_global_id("PierNode", str(pier.id))
+    berth_type_id = to_global_id("BerthTypeNode", str(berth_type.id))
+
+    variables = {
+        "number": "9999",
+        "comment": "foobar",
+        "pierId": pier_id,
+        "berthTypeId": berth_type_id,
+    }
+
+    assert len(Berth.objects.all()) == 0
+
+    executed = client.execute(
+        query=CREATE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url=GRAPHQL_URL,
+        user=superuser,
+    )
+
+    assert len(Berth.objects.all()) == 1
+    assert executed["data"]["createBerth"]["berth"]["id"] is not None
+    assert executed["data"]["createBerth"]["berth"]["comment"] == "foobar"
+    assert executed["data"]["createBerth"]["berth"]["number"] == "9999"
+    assert executed["data"]["createBerth"]["berth"]["pier"]["id"] == pier_id
+    assert executed["data"]["createBerth"]["berth"]["berthType"]["id"] == berth_type_id
+
+
+def test_create_berth_no_user(pier, berth_type):
+    variables = {
+        "number": "9999",
+        "comment": "foobar",
+        "pierId": to_global_id("PierNode", str(pier.id)),
+        "berthTypeId": to_global_id("BerthTypeNode", str(berth_type.id)),
+    }
+
+    assert len(Berth.objects.all()) == 0
+
+    executed = client.execute(
+        query=CREATE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url=GRAPHQL_URL,
+        user=None,
+    )
+
+    assert len(Berth.objects.all()) == 0
+    assert (
+        executed["errors"][0]["message"]
+        == "You do not have permission to perform this action"
+    )
+
+
+def test_create_berth_staff_user(pier, berth_type, staff_user):
+    variables = {
+        "number": "9999",
+        "comment": "foobar",
+        "pierId": to_global_id("PierNode", str(pier.id)),
+        "berthTypeId": to_global_id("BerthTypeNode", str(berth_type.id)),
+    }
+
+    assert len(Berth.objects.all()) == 0
+
+    executed = client.execute(
+        query=CREATE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url=GRAPHQL_URL,
+        user=None,
+    )
+
+    assert len(Berth.objects.all()) == 0
+    assert_not_enough_permissions(executed)
+
+
+def test_create_berth_no_number(pier, berth_type, staff_user):
+    variables = {
+        "pierId": to_global_id("PierNode", str(pier.id)),
+        "berthTypeId": to_global_id("BerthTypeNode", str(berth_type.id)),
+    }
+
+    assert len(Berth.objects.all()) == 0
```

Comment:
Just FYI you can also do this with `Berth.objects.count()` or `Berth.objects.all().count()`